lib/spareval/src/lib.rs

```diff
@@ -394,6 +394,65 @@ impl QueryEvaluator {
         self.run_stats = true;
         self
     }
+
+    // Internal helper: evaluates a SPARQL expression to an ExpressionTerm against an empty dataset
+    fn eval_expression_term_with_substitutions<'a>(
+        &self,
+        expression: &sparopt::algebra::Expression,
+        substitutions: impl IntoIterator<Item = (&'a Variable, &'a Term)>,
```

Comment:
nit: Maybe better with `Term` instead of `&Term` to avoid the clone() call (same with public APIs) but I am happy to merge the MR in the current state if you prefer.

Reply:
Implemented in https://github.com/oxigraph/oxigraph/pull/1403/commits/8b3e6eb4b99550a0b29139ae69a36450e79f2ebd